RippleCarryAdder/ReferenceImplementation.qs

```diff
@@ -288,4 +288,89 @@ namespace Quantum.Kata.RippleCarryAdder {
         // transform 2ᴺ - 1 - (b - a) into b - a
         ApplyToEachA(X, b);
     }
+    
+    //////////////////////////////////////////////////////////////////
+    // Part V. Operations Modulo 2ᴺ
+    //////////////////////////////////////////////////////////////////
+
+    // Task 5.1. Adder Modulo 2ᴺ
+    operation AdderModuloNbits_Reference (a : Qubit[], b : Qubit[], sum : Qubit[]) : Unit is Adj {
```

Comment:
You could add an assertion to check that `sum` is in the all-zero state (see `AssertAllZero`)

Reply:
Good Idea

Reply:
@msoeken This Assert actually may not be such a good idea. I added it initially as per your suggestions. However this caused the CI Build to Keep Failiing for various reasons. 

The root cause of this failure was that the test calls the operation AssertOperationEqualReferenced. This basically tests if the reference version of the operation and the user implemented version is identical. When running %check_kata it should ideally be comparing the Reference version with itself and thus be identical by definition and not fail. However presence of the AssertAllZero causes AssertOperationEqualReferenced to fail resulting in the test failing. 

Since AssertOperationEqualReferenced tests identical behaviour on all possible inputs , my guess is that when it is testing on some input where sum is not zero the assert fails and hence the test fails. This is also backed up by the fact that the AssertAllZero condition is not used by any of the previous tasks involving Addition/Subtraction.

Reply:
@msoeken After undoing this change the build again seems to be working.